--- origsrc/gelemental-1.2.0/libelemental/Makefile.am	2007-09-24 23:43:46.000000000 -0500
+++ src/gelemental-1.2.0/libelemental/Makefile.am	2007-09-30 19:28:17.437500000 -0500
@@ -6,7 +6,7 @@
 AM_CXXFLAGS = $(LIBELEMENTAL_CXXFLAGS)
 
 lib_LTLIBRARIES = libelemental.la
-libelemental_la_LDFLAGS = -version-info 0:0:0
+libelemental_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 libelemental_la_SOURCES = \
 	private.hh \
--- origsrc/gelemental-1.2.0/libelemental/misc/ucompose.cc	2007-09-24 23:39:02.000000000 -0500
+++ src/gelemental-1.2.0/libelemental/misc/ucompose.cc	2007-09-30 19:30:28.531250000 -0500
@@ -133,7 +133,7 @@
 			output.insert (pos, rep);
 		}
 
-		os.str (std::wstring ());
+		os.str (ustring ());
 		//os.clear ();
 		++arg_no;
 	}
@@ -141,7 +141,7 @@
 
 
 ustring
-UComposition::do_stringify (const std::wstring& src)
+UComposition::do_stringify (const ustring& src)
 {
 	return Glib::convert (string (reinterpret_cast<const char *> (src.data ()),
 		src.size () * sizeof (wchar_t)), "UTF-8", "WCHAR_T");
--- origsrc/gelemental-1.2.0/libelemental/misc/ucompose.hh	2007-09-24 23:39:02.000000000 -0500
+++ src/gelemental-1.2.0/libelemental/misc/ucompose.hh	2007-09-30 19:31:08.937500000 -0500
@@ -79,11 +79,11 @@
 
 	void do_arg (const ustring &rep);
 
-	ustring do_stringify (const std::wstring& src);
+	ustring do_stringify (const ustring& src);
 
 	template<typename T> inline string stringify (T obj);
 
-	std::wostringstream os;
+	std::ostringstream os;
 	int arg_no;
 
 	//!
--- origsrc/gelemental-1.2.0/src/Makefile.am	2007-09-24 23:49:18.000000000 -0500
+++ src/gelemental-1.2.0/src/Makefile.am	2007-09-30 19:37:48.609375000 -0500
@@ -13,6 +13,6 @@
 	main.hh main.cc
 
 gelemental_LDADD = \
-	$(GELEMENTAL_LIBS) \
+	../libelemental/misc/libmisc-gtk.la \
 	../libelemental/libelemental.la \
-	../libelemental/misc/libmisc-gtk.la
+	$(GELEMENTAL_LIBS)
